src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -0,0 +1,1007 @@
+    return make_document();
+}
+
+document::value next(change_stream& stream) {
+    static std::unordered_map<change_stream*, change_stream::iterator> map;
+    if (map.find(&stream) == map.end())
+        map[&stream] = stream.begin();
+
+    auto iter = map[&stream];
+    auto copy = document::value{*iter};
+    ++iter;  // prepare for next call to this function.
+    return copy;
+}
+
+document::value iterate_until_document_or_error(change_stream& stream) {
+    return make_document(kvp("result", next(stream)));
+}
+
+document::value fail_point(entity::map& map, spec::apm_checker& apm, document::view op) {
+    auto args = op["arguments"];
+    auto client_name = args["client"].get_string().value.to_string();
+    auto& client = map.get_client(client_name);
+
+    client["admin"].run_command(args["failPoint"].get_document().value);
+
+    auto event = spec::apm_checker::to_event("configureFailPoint");
+    apm.set_ignore_command_monitoring_event(event);
+
+    return make_document(kvp("uri", client.uri().to_string()),
+                         kvp("failPoint", args["failPoint"]["configureFailPoint"].get_string()));
+}
+
+document::value find_one_and_update(collection& coll, document::view operation) {
+    document::view arguments = operation["arguments"].get_document().value;
+    document::view filter = arguments["filter"].get_document().value;
+    options::find_one_and_update options{};
+
+    if (arguments["collation"]) {
+        options.collation(arguments["collation"].get_document().value);
+    }
+
+    if (arguments["hint"]) {
+        if (arguments["hint"].type() == bsoncxx::v_noabi::type::k_string)
+            options.hint(hint{arguments["hint"].get_string().value});
+        else
+            options.hint(hint{arguments["hint"].get_document().value});
+    }
+
+    if (arguments["projection"]) {
+        options.projection(arguments["projection"].get_document().value);
+    }
+
+    if (arguments["arrayFilters"]) {
+        options.array_filters(arguments["arrayFilters"].get_array().value);
+    }
+
+    if (arguments["returnDocument"]) {
+        auto return_document = arguments["returnDocument"].get_string().value.to_string();
+
+        if (return_document == "After") {
+            options.return_document(options::return_document::k_after);
+        }
+
+        if (return_document == "Before") {
+            options.return_document(options::return_document::k_before);
+        }
+    }
+
+    if (arguments["sort"]) {
+        options.sort(arguments["sort"].get_document().value);
+    }
+
+    if (arguments["upsert"]) {
+        options.upsert(arguments["upsert"].get_bool().value);
+    }
+
+    auto result = builder::basic::document{};
+    stdx::optional<document::value> document;
+
+    switch (arguments["update"].type()) {
+        case bsoncxx::type::k_document: {
+            document::view update = arguments["update"].get_document().value;
+            document = coll.find_one_and_update(filter, update, options);
```

Comment:
This should check if a session was passed (and the array case).

Reply:
Done